### ISSUE TITLE
Image Order

### ISSUE THREAD

**Issue by SamuelWei** [ivmartel/dwv#462]
Hi,

I really like dwv. My only problem is the image sorting algorithm, that is different to other dicom viewers. With my data and the order I load them via the url the order is wrong. Eg. image-id: 34->40->39

If dwv would order the images based on the number marked in the screenshot, it would work for my dicom images.
![imageorderpng](https://user-images.githubusercontent.com/4281791/37181624-1114b8a0-232e-11e8-813a-658f55ba54ec.png)

Is there a setting to change the sorting algorithm?

**Comment by tmskmr** [comment]
I agree with you, and I think that you could achieve it by customizing the funciton dwv.image.Image.appendSlice().
You need to replace  "geometry.getSliceIndex()" by the function which should return the image index based on the series number and the instance number.

**Comment by ivmartel** [comment]
dwv uses the 'Image Position Patient' DICOM field to get the coordinates of the slice. As @tmskmr says, it then guess the slice index in the GetSliceIndex at [image/geometry.js#L184](https://github.com/ivmartel/dwv/blob/v0.23.0/src/image/geometry.js#L184) ordering them via their position.

If the order is wrong, maybe the image patient field is bad... Can you check or could you share your data?

**Comment by AbdoulazizMaharou** [comment]
var globalUrl = "http://localhost/radiologie/public/uploads/";
self.loadURLs([ "http://localhost/radiologie/public/uploads/
m9FuvxIUTycoaaBR42Re.DCM","http://localhost/radiologie/public/uploads/
lOuJYhqNiMX3BckeWtNC.DCM"]);
It fonctionned with on file example: self.loadURLs([ "
http://localhost/radiologie/public/uploads/m9FuvxIUTycoaaBR42Re.DCM"]);
But if exceeded one file the follow alert dialog appears:   ​Error: Cannot
append a slice with different number of columns
 m9FuvxIUTycoaaBR42Re.DCM
<https://drive.google.com/file/d/1rJtiuNqip0uifv6D5fSds2nhqH7AlfAb/view?usp=drive_web>
​​
 lOuJYhqNiMX3BckeWtNC.DCM
<https://drive.google.com/file/d/1Nqc4WR3JRFYmaZCVZudKE2KykR9lyOeI/view?usp=drive_web>
​Thanks to you for taking your times to answer me.🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏

2018-03-12 13:02 GMT+01:00 ivmartel <notifications@github.com>:

**Comment by SamuelWei** [comment]
I rewrote the image.js file to have the slides order by the Instance Number Attribute.

[image.txt](https://github.com/ivmartel/dwv/files/1803187/image.txt)

**Comment by ivmartel** [comment]
I would still be interested if you could share the data so that I could think of a solution that uses both the image position and the instance number.

**Comment by AbdoulazizMaharou** [comment]
I have always the same error. Can you delete the comparaison ?

2018-03-12 15:38 GMT+01:00 SamuelWei <notifications@github.com>:

**Comment by AbdoulazizMaharou** [comment]
My Dicom images or the integrality of my project ??

2018-03-12 16:19 GMT+01:00 Abdoul Aziz Maharou Ado <abdoulazizado@gmail.com>
:

**Comment by SamuelWei** [comment]
[Archiv.zip](https://github.com/ivmartel/dwv/files/1803660/Archiv.zip)
I put the files in a zip. If you load the files directly, it's working fine. As soon as the files are loaded via url, like in the demo, the order is different each time. Attached are three screenshots from the first image in order. Each time I load the viewer (with the same url) it is a different one. If the image-sorting-algorythm would be working correctly I should be the same each time, that's not the case.
![test3](https://user-images.githubusercontent.com/4281791/37296795-4cb48bce-261c-11e8-9a03-8597a88b957c.jpeg)
![test2](https://user-images.githubusercontent.com/4281791/37296796-4cd1c4e6-261c-11e8-8410-8904e81cf18a.jpeg)
![test1](https://user-images.githubusercontent.com/4281791/37296797-4cedb048-261c-11e8-85ca-9047748c7678.jpeg)

**Comment by AbdoulazizMaharou** [comment]
Thank you

2018-03-12 17:39 GMT+01:00 SamuelWei <notifications@github.com>:

**Comment by ivmartel** [comment]
The problem is that your folder contains two acquisitions: an axial one (the snapshots you sent) for numbers 20:39 and a coronal one for number 0:19. dwv thinks they belong together (which is a bug) and tries to relate the different geometries which leads to random positions according to the order in which the slices are loaded. 

I have created #466 to make dwv append slices with the same image orientation. Best if you revert your change and make two groups for the data you shared.
